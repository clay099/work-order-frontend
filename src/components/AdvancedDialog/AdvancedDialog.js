import React from "react";
import {
	Dialog,
	DialogTitle,
	Button,
	DialogActions,
	DialogContent,
} from "@material-ui/core/";

const AdvancedDialog = ({
	handleClose,
	open,
	handleSubmit,
	titleText,
	id,
	buttonText,
	buttonColor,
	FormData,
	maxWidth = "sm",
}) => {
	return (
		<Dialog
			onClose={handleClose}
			open={open}
			fullWidth={true}
			maxWidth={maxWidth}
		>
			<DialogTitle id={id}>{titleText}</DialogTitle>

			<DialogContent>{FormData.map((Field) => Field)}</DialogContent>
			<DialogActions>
				<Button
					variant="outlined"
					color={buttonColor}
					onClick={handleClose}
				>
					Cancel
				</Button>
				<Button
					variant="contained"
					color={buttonColor}
					onClick={handleSubmit}
				>
					{buttonText}
				</Button>
			</DialogActions>
		</Dialog>
	);
};

export default AdvancedDialog;
