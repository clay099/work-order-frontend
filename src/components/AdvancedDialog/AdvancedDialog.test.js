import React from "react";
import { render } from "@testing-library/react";
import AdvancedDialog from "./AdvancedDialog";
console.error = jest.fn();

const AdvancedDialogParams = {
	handleClose: true,
	open: true,
	handleSubmit: true,
	titleText: "title",
	id: "test",
	buttonText: "button test",
	buttonColor: "primary",
	FormData: ["test"],
	maxWidth: "sm",
};

describe("AdvancedDialog Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<AdvancedDialog {...AdvancedDialogParams} />);
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(
			<AdvancedDialog {...AdvancedDialogParams} />
		);
		expect(asFragment).toMatchSnapshot();
	});

	test("should have button & formData in document", () => {
		const { getByText } = render(
			<AdvancedDialog {...AdvancedDialogParams} />
		);

		const button = getByText(AdvancedDialogParams.buttonText);
		expect(button).toBeInTheDocument();

		const formData = getByText(AdvancedDialogParams.FormData[0]);
		expect(formData).toBeInTheDocument();
	});
});
