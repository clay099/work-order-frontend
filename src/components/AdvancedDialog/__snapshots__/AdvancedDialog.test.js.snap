// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvancedDialog Component matches snapshot 1`] = `[Function]`;
